### PR TITLE
add post build action to record errorprone/checkstyle/findbugs on precommit java

### PR DESCRIPTION
**Pull request by adude3141**
Add another post build action to publish check style and findbugs reports.

Also added java and error prone.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by kennknowles** [comment]
Subject says "Do not merge" but I want to know "why not"? This seems great.

**Comment by adude3141** [comment]
There are several issues I have which forced me to adding that 'do not merge'
- this is using [warnings-ng-plugin](https://github.com/jenkinsci/warnings-ng-plugin) of which I am uncertain whether it is installed on apache Jenkins. Unfortunately I did not find any reference, where plugins are installed. I implemented this with [beams dockerized Jenkins](https://github.com/apache/beam/tree/master/.test-infra/dockerized-jenkins) which had it preinstalled, but [plugins.txt](https://github.com/apache/beam/blob/master/.test-infra/dockerized-jenkins/plugins.txt) does not list it. I *think* it will be available, but I do not have any clue how to verify with lack of permissions on Jenkins. Could not find a list on [apache jenkins](https://cwiki.apache.org/confluence/display/INFRA/Jenkins) Need some help here. 
- as stated by @swegner compileErrors currently still get swallowed. As error prone will end up with compile failure, we need to add this. (FindBugs does not help here, as it is never run, needs to compile first, iiuc)
- I think similar should be added to other Jenkins jobs also. e.g. python precommit, go recommit, data flow runner and similar. Of course, this might be done with a separate task.
- and unfortunately I am unsure about the workflow to get these changes in. I assume it is a merge and manual run of the seed job with fingers crossed this does not mess up the whole Jenkins setup?

Currently fiddling with second point. Not yet satisfied with my setup.

I ll ping here, as soon I did update the branch.

**Comment by adude3141** [comment]
Added java and error prone reporting.

This might be to much, but from my point of view it might be worth trying and learning if it helps and how to improve on the reports. E.g. we might aggregate these reports into a single view, drop findbugs and/or java etc...

If anyone could help or guide me, how to get those changes merged...

**Comment by mxm** [comment]
>and unfortunately I am unsure about the workflow to get these changes in. I assume it is a merge and manual run of the seed job with fingers crossed this does not mess up the whole Jenkins setup?

You can run the seed job on this PR by commenting with `Run Seed Job`.

>This might be to much

Can't hurt having more reports/information available in Jenkins, can it? 

>If anyone could help or guide me, how to get those changes merged...

I think this looks good. We just need to see why the Java PreCommit fails.

**Comment by mxm** [comment]
Run Java PreCommit

**Comment by mxm** [comment]
Run Website PreCommit

**Comment by mxm** [comment]
Run Seed Job

**Comment by adude3141** [comment]
Thx, @mxm, for taking over here.

IIUC running seed job from branch does impact the Jenkins jobs for any other PR also, right? So if someone is doing that, she has to keep a close eye one success, right?

Here, all seems to be doing well. Reports are represented as expected. Now we might discuss how to improve the current state.

Thx again!

**Comment by mxm** [comment]
Yes, running the seed job will affect the global Jenkins configuration. In case of an invalid configuration, the current configuration will not be replaced.

Thanks for taking the time to make these changes!

**Comment by adude3141** [comment]
Thx for merging!

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -761,8 +761,8 @@ class BeamModulePlugin implements Plugin<Project> {
         }
         project.tasks.withType(FindBugs) {
           reports {
-            html.enabled = true
-            xml.enabled = false
+            html.enabled = false
```
Comment (mxm) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: @adude3141 Do you think we could re-enable html output in addition to xml? It is helpful for looking at the findbugs results locally.

Comment (mxm) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: I discovered it only supports one output.

Comment (adude3141) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: @kennknowles already pushed on that on https://github.com/apache/beam/pull/7704

We should definitely change that and toggle based on environment. I ll prepare something.

Comment (mxm) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Great!

**Pull request merged**